### PR TITLE
Update Media Card to check for Supported Features

### LINKED ISSUE #4844: Media player card doesn't check supported features
With the conversion of the media player to Lit (#4761), the checks for supported features where removed.

There is a class for media player entities that has all the logic:
https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/util/hass-media-player-model.js

### PR DESCRIPTION
**Pull request by zsarnett**
## Type of change
- [x] Bugfix (non-breaking change which fixes an issue)
- [x] Code quality improvements to existing code or addition of tests

- This PR fixes or closes issue: fixes #4844

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

### CODE REVIEW

**Comment by bramkragten** [comment]
Still some big differences, mainly:
- Play button state doesn't get disabled
- Unavailable state -> Play button + image color
- Size of album covers
- The progress bar doesn't disappear when stopped playing
- Animation on height change is missing
- Missing now playing info
- Long media title isn't cut off (ellipsis)

Before:
image
media-before

After:
image
media-after

**Comment by zsarnett** [comment]
> * Play button state doesn't get disabled

Fixed. Disabled when the state is `off`

> * Unavailable state -> Play button + image color

Fixed. Play button is no longer rendered and the image is darkened when the state is `unavailable`

> * Size of album covers

This is something I changed to be consistent with the card size. I have made it so all the albums are the same size if the card is playing

> * The progress bar doesn't disappear when stopped playing

I disagree with the progress bar being removed if the state is `paused`. I would still expect the player to know my place and continue when I hit play.

> * Animation on height change is missing

Fixed. Height is animated when the image is changed

> * Missing now playing info

Fixed. This was changed the be `Secondary_Info: Primary_Info` when the card was converted. I have reverted this change.

> * Long media title isn't cut off (ellipsis)

Fixed. Media Title is not set to show ellipsis when longer than one row of text

**Comment by zsarnett** [comment]
@balloob The demo (Arsaboo) has the image pulling from api folder. The new system tries to find that and convert to base64 but for that image, in particular, it isn't found. Resulting in a blue background

**Comment by balloob** [comment]
I don't understand, we shouldn't try to fetch the image unless there is one? Also why are we converting it to base64? It's already a url. If it needs auth we should use a signed url.

**Comment by zsarnett** [comment]
The reason the image shows up blank before the recent change was that an entity image was supplied "api/.../image" URL. But that URL was not found and no image was returned. The previous way doesn't check to see if that image exists or not. We are only converting to base64 if the image is a local image.

**Comment by zsarnett** [comment]
#4489 Also Adding this here as something that may need to be checked for. Will use this PR to update

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 1]
```diff
@@ -68,28 +78,41 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
...
         >
-          <div class="image" style="background-image: url(${image})"></div>
-          <div class="caption">
+          <div
+            class="image"
+            style="background-image: url(${this.hass.hassUrl(picture)})"
+          ></div>
+          <div
+            class="${classMap({
+              caption: true,
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: Don't add static values to a `classMap`:
```
class="caption ${classMap({...})"
```

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 2]
```diff
@@ -99,35 +122,59 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
...
+                  <paper-icon-button
+                    .icon=${"hass:skip-previous"}
+                    .disabled="${OFF_STATES.includes(stateObj.state)}"
+                    .action=${"media_previous_track"}
+                    @click=${this._handleClick}
+                  ></paper-icon-button>
+                `}
+            ${(stateObj.state !== "playing" &&
+              !supportsFeature(stateObj, SUPPORTS_PLAY)) ||
+            stateObj.state === "unavailable"
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-media-control-card.ts]:
```suggestion
            stateObj.state === "unavailable" || (stateObj.state == "playing" && !supportsFeature(stateObj, SUPPORT_PAUSE) &&  !supportsFeature(stateObj, SUPPORT_STOP))
```

Comment (bramkragten) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: Use the const for `unavailable` -> https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/data/entity.ts#L1

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 3]
```diff
@@ -99,35 +122,59 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
...
+                    .icon=${"hass:skip-next"}
+                    .disabled="${OFF_STATES.includes(stateObj.state)}"
+                    .action=${"media_next_track"}
+                    @click=${this._handleClick}
+                  ></paper-icon-button>
+                `}
           </div>
           <paper-icon-button
-            icon="hass:dots-vertical"
+            .icon=${"hass:dots-vertical"}
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: Only make it a property when it dynamic

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 4]
```diff
@@ -158,32 +205,53 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
...
+    if (!newImage || newImage === oldImage) {
+      return;
+    }
+
+    if (newImage.substr(0, 1) !== "/") {
+      this._image = newImage;
+      return;
+    }
+
+    fetchMediaPlayerThumbnailWithCache(this.hass, this._config.entity).then(
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: @balloob didn't we move the thumbnails away from the ws?

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 5]
```diff
@@ -129,7 +130,7 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
             ? ""
             : html`
                 <paper-icon-button
-                  .icon=${"hass:power"}
+                  icon=${"hass:power"}
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-media-control-card.ts]:
```suggestion
                  icon="hass:power"
```

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 6]
```diff
@@ -139,15 +140,18 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
               ? ""
               : html`
                   <paper-icon-button
-                    .icon=${"hass:skip-previous"}
+                    icon=${"hass:skip-previous"}
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-media-control-card.ts]:
```suggestion
                    icon="hass:skip-previous"
```

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 7]
```diff
@@ -166,15 +170,15 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
               ? ""
               : html`
                   <paper-icon-button
-                    .icon=${"hass:skip-next"}
+                    icon=${"hass:skip-next"}
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-media-control-card.ts]:
```suggestion
                    icon="hass:skip-next"
```

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 8]
```diff
@@ -166,15 +170,15 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
...
                     .disabled="${OFF_STATES.includes(stateObj.state)}"
                     .action=${"media_next_track"}
                     @click=${this._handleClick}
                   ></paper-icon-button>
                 `}
           </div>
           <paper-icon-button
-            .icon=${"hass:dots-vertical"}
+            icon=${"hass:dots-vertical"}
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-media-control-card.ts]:
```suggestion
            icon="hass:dots-vertical"
```

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 9]
```diff
@@ -166,15 +170,15 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
...
             @click="${this._handleMoreInfo}"
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: Also we should drop the `""` on dynamic values.
```suggestion
            @click=${this._handleMoreInfo}
```

**Review by bramkragten — APPROVED** [review]

**Comment by zsarnett** [comment]
Fixes: #4503

File: src/panels/lovelace/cards/hui-media-control-card.ts (hunk 4 above)
Comment (zsarnett) [reply src/panels/lovelace/cards/hui-media-control-card.ts]: What is the status on how we should fetch the image? Should it be reverted to using the image URL or is this the best method. This is the method it was using before the conversion.

**Pull request merged**